reactive/calico.py

```diff
@@ -335,8 +340,11 @@ def registry_credentials_changed():
     for server_name, server in creds['auths'].items():
         auth = server['auth']
         username, password = b64decode(auth).decode('utf-8').split(':')
-        cmd = ['docker', 'login', server_name, '-u', username, '-p', password]
-        check_call(cmd)
+        CTL.pull(
+            server_name + CALICO_IMAGE,
+            username=username,
+            password=password
+        )
```

Comment:
We pull the `calicoctl` image here. When do we pull `cnx-node`?

Reply:
This is going to try to pull the `calicoctl` image from every registry that exists in the credentials. We should only pull from the registry that we actually use it from: https://github.com/charmed-kubernetes/layer-tigera-secure-ee/blob/1a86c1e2208b36d3d0cf74ec468a0e43b6c148a2/reactive/calico.py#L401

Reply:
> We pull the `calicoctl` image here. When do we pull `cnx-node`?

This is done from within k8s afaik; line 229.

Reply:
I don't follow. Assuming you're talking about [this line 229](https://github.com/charmed-kubernetes/layer-tigera-secure-ee/blob/1a86c1e2208b36d3d0cf74ec468a0e43b6c148a2/reactive/calico.py#L229), that function deploys a lot of components into Kubernetes, but cnx-node is not one of them.